include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -866,103 +866,35 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUImage** const _begin,
         submit_transfer.pWaitSemaphores = nullptr;
         submit_transfer.pWaitDstStageMask = nullptr;
     }
-    auto cmdUpload = [&](const asset::ICPUImage* cpuimg, IGPUImage* img) -> void
+    auto cmdUpload = [&](const asset::ICPUImage* cpuimg, IGPUImage* gpuimg) -> void
     {
-#define USE_NEW_IMAGE_UPLOAD_UTL2
-
-#if defined(USE_NEW_IMAGE_UPLOAD_UTL)
-            IGPUCommandBuffer::SImageMemoryBarrier toTransferDst = {};
-            toTransferDst.barrier.srcAccessMask = static_cast<asset::E_ACCESS_FLAGS>(0u);
-            toTransferDst.barrier.dstAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
-            toTransferDst.oldLayout = asset::IImage::EL_UNDEFINED;
-            toTransferDst.newLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
-            toTransferDst.srcQueueFamilyIndex = transferFamIx;
-            toTransferDst.dstQueueFamilyIndex = transferFamIx;
-            toTransferDst.image = core::smart_refctd_ptr<video::IGPUImage>(img);
-            toTransferDst.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT; // this probably shoudn't be hardcoded
-            toTransferDst.subresourceRange.baseMipLevel = 0u;
-            toTransferDst.subresourceRange.levelCount = img->getCreationParameters().mipLevels;
-            toTransferDst.subresourceRange.baseArrayLayer = 0u;
-            toTransferDst.subresourceRange.layerCount = cpuimg->getCreationParameters().arrayLayers;
+        IGPUCommandBuffer::SImageMemoryBarrier toTransferDst = {};
+        toTransferDst.barrier.srcAccessMask = static_cast<asset::E_ACCESS_FLAGS>(0u);
+        toTransferDst.barrier.dstAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
+        toTransferDst.oldLayout = asset::IImage::EL_UNDEFINED;
+        toTransferDst.newLayout = asset::IImage::EL_TRANSFER_DST_OPTIMAL;
+        toTransferDst.srcQueueFamilyIndex = transferFamIx;
+        toTransferDst.dstQueueFamilyIndex = transferFamIx;
+        toTransferDst.image = core::smart_refctd_ptr<video::IGPUImage>(gpuimg);
+        toTransferDst.subresourceRange.aspectMask = asset::IImage::EAF_COLOR_BIT; // this probably shoudn't be hardcoded
```

Comment:
transition all apects possible maybe?

Reply:
I think we should get that from the ICPUImage. I'll look into it